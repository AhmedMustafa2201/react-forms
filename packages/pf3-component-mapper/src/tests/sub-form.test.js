import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SubForm from '../files/sub-form';
import RenderWithProvider from '../../../../__mocks__/with-provider';

describe('<SubForm />', () => {
  let initialProps;
  let formOptions;

  beforeEach(() => {
    formOptions = {
      renderForm: ({ name }) => <div key={name}>Form item</div>
    };
    initialProps = {
      fields: []
    };
  });

  it('should render sub form with title and description', () => {
    const wrapper = shallow(
      <RenderWithProvider value={{ formOptions }}>
        <SubForm title="Foo" description="Bar" {...initialProps} />
      </RenderWithProvider>
    ).dive();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render sub form with title', () => {
    const wrapper = shallow(
      <RenderWithProvider value={{ formOptions }}>
        <SubForm title="Foo" {...initialProps} />{' '}
      </RenderWithProvider>
    ).dive();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render sub form with description', () => {
    const wrapper = shallow(
      <RenderWithProvider value={{ formOptions }}>
        <SubForm description="Bar" {...initialProps} />{' '}
      </RenderWithProvider>
    ).dive();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
