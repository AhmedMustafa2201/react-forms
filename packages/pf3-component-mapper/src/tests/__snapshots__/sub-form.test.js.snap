// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubForm /> should render sub form with description 1`] = `
<ContextProvider
  value={
    Object {
      "batch": [Function],
      "blur": [Function],
      "change": [Function],
      "destroyOnUnregister": false,
      "focus": [Function],
      "getFieldState": [Function],
      "getRegisteredFields": [Function],
      "getState": [Function],
      "initialize": [Function],
      "isValidationPaused": [Function],
      "mutators": Object {},
      "pauseValidation": [Function],
      "registerField": [Function],
      "reset": [Function],
      "resetFieldState": [Function],
      "resumeValidation": [Function],
      "setConfig": [Function],
      "submit": [Function],
      "subscribe": [Function],
    }
  }
>
  <ContextProvider
    value={
      Object {
        "formOptions": Object {
          "renderForm": [Function],
        },
      }
    }
  >
    <SubForm
      description="Bar"
      fields={Array []}
    />
     
  </ContextProvider>
</ContextProvider>
`;

exports[`<SubForm /> should render sub form with title 1`] = `
<ContextProvider
  value={
    Object {
      "batch": [Function],
      "blur": [Function],
      "change": [Function],
      "destroyOnUnregister": false,
      "focus": [Function],
      "getFieldState": [Function],
      "getRegisteredFields": [Function],
      "getState": [Function],
      "initialize": [Function],
      "isValidationPaused": [Function],
      "mutators": Object {},
      "pauseValidation": [Function],
      "registerField": [Function],
      "reset": [Function],
      "resetFieldState": [Function],
      "resumeValidation": [Function],
      "setConfig": [Function],
      "submit": [Function],
      "subscribe": [Function],
    }
  }
>
  <ContextProvider
    value={
      Object {
        "formOptions": Object {
          "renderForm": [Function],
        },
      }
    }
  >
    <SubForm
      fields={Array []}
      title="Foo"
    />
     
  </ContextProvider>
</ContextProvider>
`;

exports[`<SubForm /> should render sub form with title and description 1`] = `
<ContextProvider
  value={
    Object {
      "batch": [Function],
      "blur": [Function],
      "change": [Function],
      "destroyOnUnregister": false,
      "focus": [Function],
      "getFieldState": [Function],
      "getRegisteredFields": [Function],
      "getState": [Function],
      "initialize": [Function],
      "isValidationPaused": [Function],
      "mutators": Object {},
      "pauseValidation": [Function],
      "registerField": [Function],
      "reset": [Function],
      "resetFieldState": [Function],
      "resumeValidation": [Function],
      "setConfig": [Function],
      "submit": [Function],
      "subscribe": [Function],
    }
  }
>
  <ContextProvider
    value={
      Object {
        "formOptions": Object {
          "renderForm": [Function],
        },
      }
    }
  >
    <SubForm
      description="Bar"
      fields={Array []}
      title="Foo"
    />
  </ContextProvider>
</ContextProvider>
`;
